src/scripts/librarymenu.js

```diff
@@ -155,297 +302,671 @@ define(["dom", "layoutManager", "inputManager", "connectionManager", "events", "
+                } else {
+                    mainDrawerButton.classList.add("hide");
+                }
+            }
+
+            if ("library" !== currentDrawerType) {
+                refreshLibraryDrawer();
+            }
+        }
+
+        updateMenuForPageType(isDashboardPage, isLibraryPage);
+
+        if (!e.detail.isRestored) {
+            window.scrollTo(0, 0);
+        }
+
+        updateTitle(page);
+        updateBackButton(page);
+        updateLibraryNavLinks(page);
+    });
+
+    (function () {
+        var html = "";
+        html += '<div class="flex align-items-center flex-grow headerTop">';
+        html += '<div class="headerLeft">';
+        html += '<button type="button" is="paper-icon-button-light" class="headerButton headerButtonLeft headerBackButton hide"><i class="md-icon">' + (browser.safari ? "chevron_left" : "&#xE5C4;") + "</i></button>";
+        html += '<button type="button" is="paper-icon-button-light" class="headerButton headerHomeButton hide barsMenuButton headerButtonLeft"><i class="md-icon">&#xE88A;</i></button>';
+        html += '<button type="button" is="paper-icon-button-light" class="headerButton mainDrawerButton barsMenuButton headerButtonLeft hide"><i class="md-icon">&#xE5D2;</i></button>';
+        html += '<h3 class="pageTitle"></h3>';
+        html += "</div>";
+        html += '<div class="headerRight">';
+        html += '<span class="headerSelectedPlayer"></span>';
+        html += '<button is="paper-icon-button-light" class="headerCastButton castButton headerButton headerButtonRight hide"><i class="md-icon">&#xE307;</i></button>';
+        html += '<button type="button" is="paper-icon-button-light" class="headerButton headerButtonRight headerSearchButton hide"><i class="md-icon">&#xE8B6;</i></button>';
+        html += '<button is="paper-icon-button-light" class="headerButton headerButtonRight headerUserButton hide"><i class="md-icon">&#xE7FD;</i></button>';
+
+        if (!layoutManager.mobile) {
+            html += '<button is="paper-icon-button-light" class="headerButton headerButtonRight headerSettingsButton hide"><i class="md-icon">&#xE8B8;</i></button>';
+        }
+
+        html += "</div>";
+        html += "</div>";
+        html += '<div class="headerTabs sectionTabs hide">';
+        html += "</div>";
+        skinHeader.classList.add("skinHeader-withBackground");
+        skinHeader.innerHTML = html;
+        headerHomeButton = skinHeader.querySelector(".headerHomeButton");
+        headerUserButton = skinHeader.querySelector(".headerUserButton");
+        headerSettingsButton = skinHeader.querySelector(".headerSettingsButton");
+        headerCastButton = skinHeader.querySelector(".headerCastButton");
+        headerSearchButton = skinHeader.querySelector(".headerSearchButton");
+
+        if (!browser.chrome) {
+            skinHeader.classList.add("skinHeader-blurred");
+        }
+
+        lazyLoadViewMenuBarImages();
+        bindMenuEvents();
+    })();
```

Comment:
Not a fan of a random self-invoking function. Out of the scope of this PR though imo.